src/webapi/element.rs

```diff
@@ -42,6 +59,39 @@ pub trait IElement: IEventTarget {
             js!( return @{self.as_ref()}.querySelectorAll( @{selector} ); ).into_reference_unchecked().unwrap()
         }
     }
+
+    /// A property which represents the inner html of a element and its descendants as a DOM string.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML)
+    fn inner_html( &self ) -> String {
+        js!(
+            return @{self.as_ref()}.innerHTML;
+        ).into_string().unwrap()
+    }
+
+    /// Sets the inner html of this element. Calling this removes all
+    /// of node's children and replaces them with html elements
+    /// of the given DOM string. If this document is an XML document and you give
+    /// innerHTML an not well formed XML, this will throw an exception.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML)
+    fn set_inner_html( &self, text: &str ) -> Result< (), InvalidStateError > {
+        let status = js! {
+            try {
+                @{self.as_ref()}.innerHTML = @{text};
+                return true;
+            } catch( exception ) {
+                throw exception;
```

Comment:
You still need to check the type of the exception. (:

Reply:
Thanks! Hmm, what do you think I should do if it's not the exception type i'm expecting?

Reply:
What are you expecting and what are you getting?

Reply:
Oh! Actually, I see now what's happening in another bit of code:

This is from node.rs
if( exception instanceof NotFoundError ) {
                    return false;
} else {
        throw exception;
}

You return false and throw a correct Rust error if its what's expected, and throw a JS exception if its not what is expected (thus stopping the rust call back)